src/modules/app.ts

```diff
@@ -339,18 +339,21 @@ export class AppModule
    * @returns - The sanitized metadata with an uploaded image ipfs hash
    */
   private async _prepareMetadata(metadata: CommonModuleMetadata): Promise<any> {
-    if (typeof metadata.image === "string") {
-      return Promise.resolve(metadata);
-    }
-    if (metadata.image === undefined) {
-      return Promise.resolve(metadata);
+    for(const key in metadata) {
+      if (typeof metadata[key] === "string") { 
```

Comment:
We need an `if` block dedicated to checking if a property is  a `Dict` or `object`, in which case we need to recursively traverse that object to check its properties, and on and on.

Reply:
gotcha, but instead of creating a new breakout function why can't we just pass the dict to the `_prepareMetadata`?